compiler/front/commands.nim

```diff
@@ -1412,6 +183,184 @@ proc processArgument*(pass: TCmdLinePass; p: OptParser;
       result = true
   inc argsCount
 
+type
+  CliFlagKind* = enum
+    ## list of cli only flags
+    cliFlagVersion
+    cliFlagHelp
+    cliFlagHelpFull
+    cliFlagHelpAdvanced
+    cliFlagMsgFormat
+
+type
+  CliEventKind* = enum
+    # errors - cli command
+    cliEvtErrInvalidCommand # main.nim
+    cliEvtErrCmdMissing # cmdlinehelper.nim
+    cliEvtErrCmdExpectedNoAdditionalArgs # nim.nim
+      ## flag disallows additional args
+    cliEvtErrFlagArgExpectedFromList # commands.nim
+      ## flag expected arg from an allow/valid list, but got none
+    cliEvtErrFlagArgNotFromValidList # commands.nim
+      ## flag expects arg from an allow/valid list
+    cliEvtErrRunCmdFailed  # commands.nim and nim.nim
+    cliEvtErrGenDependFailed # main.nim
+    # errors - general flag/option/switches (TODO: standardize on "flag")
+    cliEvtErrUnexpectedRunOpt # commands.nim and nim.nim
+    cliEvtErrFlagArgForbidden # help/version/fullhelp/etc disallow args
+    cliEvtErrFlagProcessing # commands.nim
+    # errors - specific flag/option/switch
+    cliEvtErrNoCliParamsProvided # commands.nim and nim.nim
+    # warnings - general flags/options/switches
+    cliEvtWarnSwitchValDeprecatedNoop
+    # hints - general flag/options/switches and/or processing
+    cliEvtHintPathAdded
+      ## currently only triggered if nimble adds a path
+      # xxx: this doesn't feel like a hint, more like "info" or "trace"
+
+  CliEvent* = object
+    # xxx: should these be 'flat' log events? put another way is the 
+    #      `procResult` field a design smell?
```

Comment:
I personally think the nested approach (i.e., storing `ProcSwithResult` here) is fine. The subset-of relationship between switch processing error and CLI errors is made clear, and it also allows for re-using the `ProcSwitchResult` rendering facilities.

Reply:
Thanks, I rewrote the comment based upon yours.